lib/sycamore/sycamore/utils/deep_eq.py

```diff
@@ -0,0 +1,49 @@
+from sycamore.data import Element
+
+
+def assert_deep_eq(a, b, path):
+    assert type(a) == type(b), f"type {a} {b} {path}"
+    if a is None:
+        assert b is None
+        return True
+
+    if isinstance(a, int) or isinstance(a, float) or isinstance(a, str):
+        assert a == b, f"values {a} {b} at {path}"
+        return True
+
+    if isinstance(a, list) or isinstance(a, tuple):
+        assert len(a) == len(b), f"length {len(a)} {len(b)} {path}"
+        for i, v in enumerate(a):
```

Comment:
You're not using v.  Either change a[i] to v or make the loop over range(len(a)).

Reply:
Done.